### PR TITLE
Support domains and better error handling

### PR DESCRIPTION
**Pull request by tjmehta**

### CODE REVIEW

**Comment by tjmehta** [comment]
@cflyn07

**Comment by tjmehta** [comment]
@cflynn07

File: index.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ require('methods').forEach(function (method) {
     opts.method = method.toUpperCase();
     args = [opts, cb].filter(exists);
     var req = createReq(this.app, opts);
-    var res = createRes(this.app, opts, cb, args);
+    var res = createRes(this.app, opts, cb, process.domain);
```
Comment (cflynn07) [inline index.js]: Is this used in `createRes`

File: index.js [hunk 2]
```diff
@@ -142,7 +144,24 @@ function createRes (app, opts, cb) {
...
+    function send () {
+      sent = true;
+      // next tick it to avoid express catching runtime error in the callback
+      //   getting caught by express and triggering app's error handler
+      process.nextTick(function () {
+        cb(null, res, body);
+      });
+    }
   };
   // }
```
Comment (cflynn07) [inline index.js]: remove

**Pull request merged**